pysmt/solvers/cvc4.py

```diff
@@ -246,11 +246,13 @@ def walk_ite(self, formula, args, **kwargs):
     def walk_real_constant(self, formula, **kwargs):
         frac = formula.constant_value()
         n,d = frac.numerator, frac.denominator
-        return self.mkConst(CVC4.Rational(n, d))
+        rep = str(n) + "/" + str(d)
+        return self.mkConst(CVC4.Rational(rep))
 
     def walk_int_constant(self, formula, **kwargs):
-        assert type(formula.constant_value()) == int
-        return self.mkConst(CVC4.Rational(formula.constant_value()))
+        assert is_pysmt_integer(formula.constant_value())
+        rep = str(formula.constant_value())
+        return self.mkConst(CVC4.Rational(rep))
```

Comment:
Yes. You can build an Integer object, but the ```mkConst``` is not defined for that case. I also had the same doubt and initially wrote Integer, but it was failing.